build/build.go

```diff
@@ -225,7 +225,10 @@ func parseAndAugment(xctx XContext, pkg *PackageData, isTest bool, fileSet *toke
 			if err != nil {
 				panic(err)
 			}
-			file, err := parser.ParseFile(fileSet, fullPath, r, parser.ParseComments)
+			// Files should be uniquely named and in the original package directory in order to be
+			// ordered correctly
+			newPath := path.Join(pkg.Dir, "__"+name)
```

Comment:
Let's use `gopherjs__` prefix to make sure there is absolutely no chance of a name clash with upstream files. 

Also IIRC Go build tooling treats files with names starting with `_` as special (for example, such directories will not be considered as packages), so I would avoid using such prefixes just in case it breaks some deeply baked-in assumptions in one if the `go/...` libraries.